src/xcode/ENA/ENA/Source/Client/HTTP Client/HTTPClient.swift

```diff
@@ -128,7 +128,9 @@ final class HTTPClient: Client {
 		completion completeWith: @escaping HourCompletionHandler
 	) {
 		let url = configuration.diagnosisKeysURL(day: day, hour: hour, forCountry: country)
-
+		let config = URLSessionConfiguration.coronaWarnSessionConfigurationWifiOnly()
+		session.configuration.allowsConstrainedNetworkAccess = config.allowsConstrainedNetworkAccess
```

Comment:
Please correct me if I'm wrong, but here you are changing the session settings of the HTTPClient.
This settings are never changed back.
In the next request (also a day request) the wifi settings will be used.
The HTTPClient is only initialized once.

Reply:
Shall we reset the configuration after loading the hour packages?

Reply:
Nice catch Artur.
IMHO we should have two different Sessions. This would allow us later to have the two downloads run in parallel and we don't have to reset anything after the hour packages were downloaded.

Reply:
Hmm I'm not sure if this is a good solution.
Because the requests are dispatched async and another request (maybe a day request) could be dispatched (with the wrong session config) while we are waiting for the hour request to come back.

Reply:
Solutions could be: 

- We use 2 separate sessions.
- We reconfigure the session for every request.

@marcussc  Any other suggestions?

Reply:
@ArturFriesen 
Nope but I would prefer 2 separate sessions for the reasons you stated. :)

Reply:
I put this already on my list. The Client is currently very inflexible. We already have a dedicated `CachingHTTPClient` to handle similar issues in other places. Maybe we create another WiFiOnlyClient for the moment and refactor them altogether?